### PR TITLE
Remove import Evaluator

### PR DESCRIPTION
**Pull request by johnbarUW**
Working Remove Unused Import Evaluator.

A new listener was created to watch for any Annotation Model changes. Upon any annotation changes, then the Remove Import Evaluator will check all annotations.

We still can't (or haven't found a way) to use IMarkers for a more thorough match, since IMarkers and document annotations aren't synchronized. 

Notes:
We probably should figure out our naming scheme. I would vote for "Listener" over "Tracker", but as long as we're consistent it doesn't really matter.

### CODE REVIEW

File: backend_plugin/src/evaluators/Evaluator.java [hunk 1]
```diff
@@ -22,10 +26,15 @@
 	// TODO: Create an interface / abstract class for different evaluation functions
```
Comment (DavidThien) [inline backend_plugin/src/evaluators/Evaluator.java]: Added an issue to keep track of this todo [https://github.com/DavidThien/IDE-IT/issues/13](https://github.com/DavidThien/IDE-IT/issues/13)

File: backend_plugin/src/main/evaluators/Evaluator.java [hunk 2]
```diff
@@ -57,22 +68,37 @@ private void initializeListeners(ITextEditor textEditor) {
 		DocumentChangesTracker docTracker = new DocumentChangesTracker(this);
 		doc.addDocumentListener(docTracker);
 		this.documentChangesTracker = docTracker;
+		
+		// Add a AnnotationModelListener to the AnnotationModel
+		AnnotationModelListener annoTracker = new AnnotationModelListener(this);
```
Comment (DavidThien) [inline backend_plugin/src/main/evaluators/Evaluator.java]: We call this variable 3 different things in 4 lines, it is an `AnnotationModelListener`, named `annoTracker`, and `annoChangesTracker` which is a bit confusing. I would suggest calling both the variables `annotationListener` or something along those lines for clarity.

Comment (DavidThien) [inline backend_plugin/src/main/evaluators/Evaluator.java]: We also create a new local variable when we can directly assign the new `AnnotationModelListener` to the instance variable.

**Review by DavidThien — CHANGES_REQUESTED** [review]
Looks good, just a minor change in one of the files.

**Review by wahleric — CHANGES_REQUESTED** [review]
I agree with David's comment about the 4 lines in Evaluator.java. I think adding a few more comments in those lines might help to understand exactly what is going on

**Comment by johnbarUW** [comment]
Good catch on those variable names. Those were a mess.  I added a couple comments for clarification as well.

**Review by DavidThien — APPROVED** [review]

**Review by wahleric — APPROVED** [review]
Looks good

**Pull request merged**